cherrypy/_cpcompat.py

```diff
@@ -21,6 +21,43 @@
 import http.client
 
 
+def _cgi_parseparam(s):
+    while s[:1] == ';':
+        s = s[1:]
+        end = s.find(';')
+        while end > 0 and (s.count('"', 0, end) - s.count('\\"', 0, end)) % 2:
+            end = s.find(';', end + 1)
+        if end < 0:
+            end = len(s)
+        f = s[:end]
+        yield f.strip()
+        s = s[end:]
+
+
+def cgi_parse_header(line):
```

Comment:
I know @webknjaz said to put this in compat, but it really doesn't belong in compat. The moment this PR is merged, it's no longer there for compatibility, but is core functionality. I'll move it.

Reply:
Done in dcefcbe9.